crates/sui-indexer/src/apis/read_api_v2.rs

```diff
@@ -103,7 +155,32 @@ impl ReadApiServer for ReadApiV2 {
         limit: Option<usize>,
         descending_order: bool,
     ) -> RpcResult<CheckpointPage> {
-        unimplemented!()
+        let cursor = cursor.map(BigInt::into_inner);
+        let limit = sui_json_rpc::api::validate_limit(
+            limit,
+            sui_json_rpc::api::QUERY_MAX_RESULT_LIMIT_CHECKPOINTS,
+        )
+        .map_err(SuiRpcInputError::from)?;
+
+        let mut checkpoints = self
+            .inner
+            .spawn_blocking(move |this| this.get_checkpoints(cursor, limit + 1, descending_order))
+            .await?;
+
+        let has_next_page = checkpoints.len() > limit;
+        checkpoints.truncate(limit);
+
+        let next_cursor = if has_next_page {
```

Comment:
`next_cursor` should point to the last elment in the truncated list even if there is no next page, b/c
1. it's an exclusive cursor so the client will not read repeated items if there is no more new item
2. clients can count on this cursor to iterate to next page even if they hit the latest page. Otherwise, the clients would hit the latest page and immediately restart from the first page.